app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,15 @@
+###
+#  Controller for handling login/logout and redirecting with flash messages.
+###
+class AuthenticationController < ApplicationController
+  def login
+    session[:sunet] = request.env['REMOTE_USER'] || ENV['REMOTE_USER']
+    flash[:success] = 'You have been successfully logged in.'
+    redirect_to params[:referrer] || :back
+  end
+
+  def logout
+    session[:sunet] = nil
+    redirect_to 'http://weblogin.stanford.edu/logout'
```

Comment:
Hm, I haven't seen this pattern for logging out in a rails app.  Typically, we configure a local endpoint (in apache) that logs the user out.  Perhaps @eefahy or @tallenaz can help get apache configured to log users out (which might also be easier to migrate once WebAuth is EOLed).